client/asset/btc/btc.go

```diff
@@ -1747,336 +1733,206 @@ func (btc *ExchangeWallet) LocktimeExpired(contract dex.Bytes) (bool, time.Time,
+	//     by the block monitoring loop.
+	//  2) A reorg happens, and this tip becomes orphaned. In this case, the
+	//     worst that can happen is that a shorter chain will replace a longer
+	//     one (extremely rare). Even in that case, we'll just log the error and
+	//     exit the block loop.
+	tipHeight, err := btc.node.getBestBlockHeight()
+	if err != nil {
+		btc.log.Errorf("tryRedemptionRequests getBestBlockHeight error: %v", err)
+		return
 	}
 
-	btc.log.Debugf("%d redemptions found, %d canceled out of %d contracts in mempool",
-		totalFound, totalCanceled, contractsCount)
-}
-
-// findRedemptionsInBlockRange attempts to find spending info for the specified
-// contracts by searching every input of all txs in the provided block range.
-// If spending info is found for any contract, the contract is purged from the
-// findRedemptionQueue and the contract's secret (if successfully parsed) or any
-// error that occurs during parsing is returned to the redemption finder via the
-// registered result chan.
-// Also checks mempool for potential redemptions if spending info is not found
-// for any of these contracts in the specified block range.
-func (btc *ExchangeWallet) findRedemptionsInBlockRange(startBlock, endBlock *block, contractOutpoints []outPoint) {
-	contractsCount := len(contractOutpoints)
-	btc.log.Debugf("finding redemptions for %d contracts in blocks %d - %d",
-		contractsCount, startBlock.height, endBlock.height)
+	iHash, iHeight := lowestBlock()
 
-	nextBlockHash := startBlock.hash
-	var lastScannedBlockHeight int64
-	var totalFound, totalCanceled int
-
-rangeBlocks:
-	for nextBlockHash != "" && lastScannedBlockHeight < endBlock.height {
-		blk, err := btc.getVerboseBlockTxs(nextBlockHash)
+	// If a startBlock is provided at a higher height, use that as the starting
+	// point.
+	if startBlock != nil {
+		h, err := btc.node.getBlockHeight(startBlock)
 		if err != nil {
-			// Redemption search for this set of contracts is compromised. Notify
-			// the redemption finder(s) of this fatal error and cancel redemption
-			// search for these contracts. The redemption finder(s) may re-call
-			// btc.FindRedemption to restart find redemption attempts for any of
-			// these contracts.
-			err = fmt.Errorf("error fetching verbose block %s: %w", nextBlockHash, err)
-			btc.fatalFindRedemptionsError(err, contractOutpoints)
+			btc.log.Errorf("tryRedemptionRequests getBlockHeight error: %v", err)
 			return
 		}
-		scanPoint := fmt.Sprintf("block %d", blk.Height)
-		lastScannedBlockHeight = int64(blk.Height)
-		for t := range blk.Tx {
-			tx := &blk.Tx[t]
-			found, canceled := btc.findRedemptionsInTx(scanPoint, tx, contractOutpoints)
-			totalFound += found
-			totalCanceled += canceled
-			if totalFound+totalCanceled == contractsCount {
-				break rangeBlocks
-			}
+		// If the startBlock height is lower than the lowest swap block height,
+		// ignore the startBlock.
```

Comment:
Don't see a reason why we can't just use startBlock directly without checking for the lowest swap block.

- When called from `btc.FindRedemption`, there's just one request and the lowest swap block is the block in which the single contract is mined. Can just pass that block in as the startBlock. If unmined, that block is nil and blocks search should be skipped. Currently, the genesis block will be searched if the single request passed to this method is for an unmined contract.

- When called from the block monitoring loop, a start block hash is provided, which is the hash of the latest block. The only time this start block will not be used is if it is lower than the lowest swap block height, as commented here. I expect the best block height would never be lower than the block height of any swap. Or is there a way this could happen?